scipy/signal/_signaltools.py

```diff
@@ -527,6 +551,11 @@ def _freq_domain_conv(in1, in2, axes, shape, calc_fast_len=False):
     else:
         fft, ifft = sp_fft.fftn, sp_fft.ifftn
 
+    if xp.isdtype(in1.dtype, 'integral'):
+        in1 = xp.astype(in1, xp.float64)
+    if xp.isdtype(in2.dtype, 'integral'):
+        in2 = xp.astype(in2, xp.float64)
```

Comment:
does this PR maintain dtype promotion behaviour for e.g. int32 input?

Reply:
I believe so, yes.